boot/zephyr/main.c

```diff
@@ -178,16 +178,22 @@ static void do_boot(struct boot_rsp *rsp)
     /* Get ram address for image */
     vt = (struct arm_vector_table *)(rsp->br_hdr->ih_load_addr + rsp->br_hdr->ih_hdr_size);
 #else
-    uintptr_t flash_base;
     int rc;
+    const struct flash_area *fap;
+    int area_id;
+    uint32_t dst[2];
 
     /* Jump to flash image */
-    rc = flash_device_base(rsp->br_flash_dev_id, &flash_base);
+    area_id = flash_area_id_from_image_slot(0);
```

Comment:
Looks like this breaks XIP

Reply:
Yes it does, because it hardcodes vt to primary slot.

Reply:
I am unclear how this breaks XIP.

Reply:
Is'nt that what we currently do i.e boot with respect to flash_base.

Reply:
flash_base is set to the slot's base address, that can be slot 0 or slot 1, here you are forcing it to slot 0. See `rsp->br_flash_dev_id = flash_area_get_device_id(BOOT_IMG_AREA(state, active_slot));`

Reply:
Currently flash_base is always set to CONFIG_FLASH_BASE_ADDRESS:
https://github.com/mcu-tools/mcuboot/blob/main/boot/zephyr/flash_map_extended.c#L35

Reply:
That is not related to `rsp->br_image_off`

Reply:
@nordicjm I am not understanding the above comment.

The current implementation will always read the Zephyr binary with address `CONFIG_FLASH_BASE_ADDRESS` as the starting point. 
This PR adds more flexibility on the starting address where to look for Zephyr binary.

Reply:
>  This PR adds more flexibility on the starting address where to look for Zephyr binary.

No, it completely breaks XIP, that is not more flexible. I've pointed this out many times now, you need to follow the code. Old code:
```
    rsp->br_image_off = boot_img_slot_off(state, active_slot);
    rc = flash_device_base(rsp->br_flash_dev_id, &flash_base);
    vt = (struct arm_vector_table *)(flash_base +
                                     rsp->br_image_off +
                                     rsp->br_hdr->ih_hdr_size);
```
So yes, the base flash address *which I never ever mentioned* is the same, the image offset which would point to the base **of the selected image** is now completely absent from this code:
```
    area_id = flash_area_id_from_image_slot(0);
    rc = flash_area_read(fap, rsp->br_hdr->ih_hdr_size, dst, 8);
    vt = (struct arm_vector_table *)dst;
```

Reply:
@nordicjm @de-nordic @d3zd3z I have pushed an update to the implementation after understanding the XIP issues that were highlighted by the prior implementation.

Reply:
@nordicjm @de-nordic @d3zd3z can you help re-review.